src/CLRunner.ts

```diff
@@ -1060,20 +1060,20 @@ export class CLRunner {
                 }
                 else {
                     try {
+                        // create a copy of the map before calling into logic api
```

Comment:
Might be good to have a comment here explaining the workaround for future reference